### DISCUSSION TITLE
Impute Gene Score matrix per chromosome

### DISCUSSION THREAD

**Discussion by jindalk** [GreenleafLab/ArchR#1575] [Questions / Documentation]
Hi, I am trying to run imputation on a large dataset and have been running into memory issues. One work around I have thought of is to obtain GSM matrices by chromosome, impute them in parallel and then merge them with cbind. Is that equivalent to running imputation on the entire matrix? I was just not sure if ArchR performs any scaling as part of the imputation function which might change my results if i am running it only on a subset of features at a time.

Thanks

**Comment by rcorces** [comment]
Once you run `addImputeWeights()`, the weights are stored per cell in your ArchRProject.  I believe those weights could be applied using `imputeMatrix()` on matrix chunks without any problem. You could check by testing on a single gene.

**Reply by jindalk** [reply]
Great! Yes that's exactly what I tried, loaded chunks of the GSM matrix (using the seqnames argument) and then imputed them using `imputeMatrix()`

Thanks
